test/api/backoff_test.dart

```diff
@@ -14,27 +14,28 @@ Future<Duration> measureWait(Future<void> future) async {
   return clock.now().difference(start);
 }
 
-void main() {
-  test('BackoffMachine timeouts are random from zero to 100ms, 200ms, 400ms, ...', () {
+/// Verify that a list of randomly generated duration will very likely be less
+/// than the specified maximum values.
+///
+/// [computeActualDurations] returns the a list of actual durations
+/// non-deterministically, corresponding to [expectedMaxDurationsInMs].
+void verifyRandomizedDurations({
+  required List<int> expectedMaxDurationsInMs,
+  required List<Duration> Function(List<Duration>) computeActualDurations,
+}) {
     // This is a randomized test.  [numTrials] is chosen so that the failure
-    // probability < 1e-9.  There are 2 * 11 assertions, and each one has a
-    // failure probability < 1e-12; see below.
+    // probability < 1e-9 for up to 1000 assertions.
+    // There are 2 * [expectedMaxDurationsInMs.length] assertions,
+    // and each one has a failure probability < 1e-12; see below.
```

Comment:
The part "for up to 1000 assertions" is new in this commit. What does it mean, and is there a logic change it corresponds to?

Reply:
This commit allows an arbitrary number of expected durations to be passed, so the `1000` is here to convey how many expected values can be passed for the upper bound of 1e-9 to still hold.